cannonfile.toml

```diff
@@ -0,0 +1,67 @@
+name = 'cartesi-rollups'
+description = 'Smart Contracts for Cartesi Rollups'
+version = '2.0.0'
+
+[deploy.InputBox]
+artifact = 'InputBox'
+create2 = true
+ifExists = 'continue'
```

Comment:
Why do you need all these ifExists?

Reply:
If, for some reason, `InputBox` is already deployed before running the script, the whole script doesn't fail, and instead just carries on. Because of the hash collision resistance properties of `keccak256` and how addresses are calculated with `CREATE2`, we can assume that if the pre-calculated address contains code, it is the same code that we're trying to deploy.

Reply:
I think that has more potential to cause confusion than help users.
For example, some may think that they should run a load_state and then run cannon, which is not the expected process.

Reply:
I don't see how setting `ifExists` may lead someone into thinking that this is the expected process. Also, to have a contract already deployed on a network before running the script is something perfectly possible to have in a real network, not just an Anvil private node. If anything, I believe this option only makes the Cannonfile more robust.

Reply:
Another possibility is that we release a major version which changes some contracts, but not all of them. So, for example, we change the `ERC20Portal` but not the `InputBox` contract, we still want the script to carry on if there is already an `InputBox` contract deployed, right?

Reply:
> So, for example, we change the `ERC20Portal` but not the `InputBox` contract, we still want the script to carry on if there is already an `InputBox` contract deployed, right?

In that case it's another version, say 1.1.0, and in that you can define the InputBox as ifExists. But not in the 1.0.0 version.

I don't see why an error of running the 1.0.0 when it's already deployed is a problem.

Reply:
> I don't see why an error of running the 1.0.0 when it's already deployed is a problem, or beneficial.

It's a problem, because the script is failing when, in fact, it shouldn't. It is fine to have a contract already deployed at that address, because we are using `CREATE2`.